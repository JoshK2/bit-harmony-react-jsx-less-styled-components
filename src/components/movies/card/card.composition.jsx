import React from 'react';
import { Card } from './card';

export const MovieCardExample = () => {
  return (
    <Card
      title="The Wolf of Wall Street"
      imdbID="tt0993846"
      poster="https://m.media-amazon.com/images/M/MV5BMjIxMjgxNTk0MF5BMl5BanBnXkFtZTgwNjIyOTg2MDE@._V1_SX300.jpg"
    />
  );
};
