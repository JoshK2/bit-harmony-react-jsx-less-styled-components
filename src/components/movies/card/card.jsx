import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-modal';
import { Button } from '../../base-ui/button';
import { MovieDetails } from '../../forms/movie-details';
import styles from './card.module.less';

const customStyles = {
  content: {
    top: '50%',
    left: '50%',
    right: '50%',
    bottom: 'auto',
    marginRight: '-50%',
    transform: 'translate(-50%, -50%)',
  },
};

export const Card = ({ title, imdbID, poster }) => {
  const [modalIsOpen, setIsOpen] = useState(false);

  return (
    <>
      <div className={styles.container}>
        <img className={styles.image} src={poster} alt={title} />
        <div className={styles.title}>{title}</div>
        <Button className={styles.button} onClick={() => setIsOpen(true)}>
          More Info
        </Button>
      </div>
      <Modal
        isOpen={modalIsOpen}
        onRequestClose={() => setIsOpen(false)}
        contentLabel="Movie Details"
        style={customStyles}
      >
        <MovieDetails imdbID={imdbID} />
      </Modal>
    </>
  );
};

Card.propTypes = {
  /** Movie title */
  title: PropTypes.string.isRequired,
  /** The movie ID from the API */
  imdbID: PropTypes.string.isRequired,
  /* Movie image */
  poster: PropTypes.string.isRequired,
};
