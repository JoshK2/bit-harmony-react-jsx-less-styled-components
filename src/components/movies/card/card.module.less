.container {
  margin: 15px;
  width: 300px;
  box-shadow: 10px 10px 23px -17px rgba(0, 0, 0, 0.75);
}
.image {
  width: 300px;
  height: 300px;
  margin-bottom: 10px;
}
.title {
  font-size: 18px;
  font-weight: bold;
  margin-bottom: 10px;
}
.button {
  margin-bottom: 15px;
}
