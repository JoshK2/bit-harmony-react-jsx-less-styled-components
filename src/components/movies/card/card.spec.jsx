import React from 'react';
import testRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { MovieCardExample } from './card.composition';

describe('Movie card component', () => {
  it('renders correctly', () => {
    testRenderer.create(<MovieCardExample />);
  });
  it('snapshot test', () => {
    const component = testRenderer.create(<MovieCardExample />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with value', () => {
    const { getByText } = render(<MovieCardExample />);
    expect(getByText(/The Wolf of Wall Street/)).toBeTruthy();
  });
});
