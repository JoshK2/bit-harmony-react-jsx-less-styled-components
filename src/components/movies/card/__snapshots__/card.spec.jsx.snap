// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie card component snapshot test 1`] = `
<div>
  <img
    alt="The Wolf of Wall Street"
    src="https://m.media-amazon.com/images/M/MV5BMjIxMjgxNTk0MF5BMl5BanBnXkFtZTgwNjIyOTg2MDE@._V1_SX300.jpg"
  />
  <div>
    The Wolf of Wall Street
  </div>
  <button
    className="sc-bdfBwQ kvbBrN"
    onClick={[Function]}
  >
    More Info
  </button>
</div>
`;
